### PR TITLE
Angus/hips remote image load

### LINKED ISSUE #1379: Support for loading HiPS FITS images
Load FITS images from hips2fits server by URL.  Using cfitsio fits_open_file, files can be opened over the network using FTP or HTTP protocols by supplying the appropriate URL as the filename.

New protobuf messages REMOTE_FILE_REQUEST and REMOTE_FILE_RESPONSE (containing OPEN_FILE_ACK) will be added to carta-protobuf to support this feature.  The request fields will be used to form the proper URL.

### PR DESCRIPTION
**Pull request by veggiesaurus**
**Description**
Load files from the hips2fits server

* What is implemented or fixed? Mention the linked issue(s), if any.
Closes #1379 

* How does this PR solve the issue? Give a brief summary.
Use CartaFitsLoader to open the file using the URL as the filename.

* Are there any companion PRs (frontend, protobuf)?
CARTAvis/carta-frontend#2346, CARTAvis/carta-protobuf#93

* Is there anything else that testers should know (e.g. exactly how to reproduce the issue)?
Load HiPS FITS image using the frontend data query UI.

**Checklist**

- [x] changelog updated / ~no changelog update needed~
- [x] e2e test passing / corresponding fix added / new e2e test created
- [ ] ICD test passing / corresponding fix added / new ICD test created
- [x] protobuf updated to the latest dev commit / ~no protobuf update needed~
- [x] protobuf version bumped / ~protobuf version not bumped~
- [x] added reviewers and assignee
- [ ] added ZenHub estimate, milestone, and release

### CODE REVIEW

**Comment by veggiesaurus** [comment]
@pford Do you think this is on the right track? this is for loading HIPS files from the hips2fits service using cfitsio's ability to load from URL. Some of the changes are required to avoid opening files multiple times

@confluence can we assign this to you or @confluence for cleanup/testing once the frontend is fleshed out? (it's only scripting at the moment)

**Comment by kswang1029** [comment]
@pford this PR looks fine to me. Would you turn this as a formal one when you feel ready?

File: src/Util/File.cc [hunk 1]
```diff
@@ -34,6 +34,10 @@ bool IsCompressedFits(const std::string& filename) {
     return false;
 }
 
+bool IsRemoteHttpFile(const std::string& filename) {
+    return filename.find("http") == 0;
+}
+
```
Comment (confluence) [inline src/Util/File.cc]: I know that this is only called after other file options have been tried, but I would prefer to make this less permissive by explicitly checking for a full `http://` or `https://` prefix using a regex, e.g.:
```
const std::regex is_http_url("^https?://");
return std::regex_search(f, is_http_url)
```

**Review by confluence — APPROVED** [review]

**Review by kswang1029 — APPROVED** [review]
looks good. e2e tests are adjusted accordingly and a new e2e test is added. 👍

**Review by acdo2002 — APPROVED** [review]
Now the AppImage package problem has been fixed, then I approve this PR.

**Pull request merged**